src/test/java/org/broadinstitute/hellbender/utils/UtilsUnitTest.java

```diff
@@ -613,6 +613,75 @@ public void testLastIndexOfLastBoundaries() {
         Assert.assertEquals(result, expected);
     }
 
+    @Test
+    public void testLastIndexOfOneMismatchLastBoundaries() {
+        final String reference = "AAAACCCCTTTTGGGG";
+
+        // match right boundary of reference
+        String query = "TGAGG";
+        int result = Utils.lastIndexOfOneMismatch(reference.getBytes(), query.getBytes());
+        int expected = 11;
+        Assert.assertEquals(result, expected);
+
+        // match left boundary of reference
+        query = "AAGAC";
+        result = Utils.lastIndexOfOneMismatch(reference.getBytes(), query.getBytes());
+        expected = 0;
+        Assert.assertEquals(result, expected);
+    }
+
+    @Test
+    public void testLastIndexOfOneMismatchRandom() {
```

Comment:
You should also test the cases 1) an exact match, 2) two mismatches i.e. query = AACCGG reference = AATTGG, and 3) one mismatch occurs more than once.

Reply:
what do you mean by "one mismatch occurs more than once? How is that different than two mismatches?

Reply:
Suppose the query is 20 bases and reference bases 21-40 and reference bases 81-100 each match the query with one mismatch (as opposed to: "reference bases 61-80 match with two mismatches").